### ISSUE TITLE
Symbols appear as streaks with usetex=True, times font and PDF backend

### ISSUE THREAD

**Issue by pdhall99** [matplotlib/matplotlib#11259]
### Bug report

**Bug summary**

- Symbols appear as streaks with usetex=True, times font and PDF backend

**Code for reproduction**
```python
import matplotlib as mpl
import matplotlib.pyplot as plt

mpl.rcParams.update(mpl.rcParamsDefault)
mpl.rcParams['text.usetex'] = True
mpl.rcParams['font.family'] = ['serif']
mpl.rcParams['font.serif'] = ['times']

fig, ax = plt.subplots()
ax.set_xlabel(r'$x$ $\alpha$')
fig.savefig('test.pdf')
```

**Actual outcome**
[test.pdf](https://github.com/matplotlib/matplotlib/files/2009824/test.pdf)

- Alpha symbol in the x-axis label appears as a streak in the output PDF

**Expected outcome**

- Alpha symbol in the x-axis label to appear as an alpha, rendered by LaTeX

**Matplotlib version**

- Matplotlib 2.2.2, Python 2.7.15, OSX 10.13.4
- pdfTeX 3.14159265-2.6-1.40.19 (TeX Live 2018), kpathsea version 6.3.0
- Installed with MacPorts

**Comment by jklymak** [comment]
I can't reproduce MacOS 10.13.4 py 3.6, conda, 
```
$ latex --version
pdfTeX 3.14159265-2.6-1.40.18 (TeX Live 2017)
kpathsea version 6.2.3
```

**Comment by jklymak** [comment]
Oddly, your file is 10 bytes larger than mine.  Otherwise I don't see a difference.

**Comment by jklymak** [comment]
Agreed.  Its strange I can't replicate it - I doubt the latex versions are the issue.  

Did you clear your cache: (on my machine `~/.matplotlib/tex.cache`) ?  Maybe you had an old alpha in there causing grief?

**Comment by pdhall99** [comment]
Thanks. I have tried clearing the cache, but it didn't help.

I am not sure how to use the contents of tex.cache to find clues. Are you able to advise?

**Comment by pdhall99** [comment]
The alpha dvi in tex.cache looks fine.

**Comment by jklymak** [comment]
What about the png?

**Comment by pdhall99** [comment]
There is no png. Should there be?

**Comment by jklymak** [comment]
Oh, maybe not if you just use the PDF backend.  I assume if you add a `plt.show()` the GUI plot looks OK?  

Are you 100% sure you are using 2.2.2?  i.e. can you add 
```python
import matplotlib
print(matplotlib.__version__)
``` 
to the above script?  

Its strange that our almost-identical setups would behave differently.  I suppose I could update TexLive just to be 100% certain...

**Comment by pdhall99** [comment]
The GUI plot looks OK when I add `plt.show()`, but I get a message:
```
/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/matplotlib/font_manager.py:1328: UserWarning: findfont: Font family [u'serif'] not found. Falling back to DejaVu Sans
  (prop.get_family(), self.defaultFamily[fontext]))
```
I am unsure how relevant this is, because the GUI plot and PDF show Times not DejaVu Sans for the tick labels?

I am using matplotlib 2.2.2. I added the lines you suggested and the output is 2.2.2

**Comment by pdhall99** [comment]
I think the warning is because I only have Times.ttc on my machine, not Times.ttf. I don't think it is relevant to my issue.

**Comment by pharshalp** [comment]
I can reproduce this issue on Mac OS 10.13.4, Python 3.6.5 :: Anaconda custom (64-bit), Matplotlib 2.2.2, pdfTeX 3.14159265-2.6-1.40.19 (TeX Live 2018)

**Comment by jklymak** [comment]
- Can anyone produce this *not* on macOS? 
 - Is the above code the *minimal* code needed to cause the error?  i.e. do you need to specify `font-family` and `font.serif`?  I find needing these a bit strange...
 - Do other greek symbols have a problem?  
 - Is it just the PDF backend w/ the problem or does svg and eps show this too?

**Comment by jkulesza** [comment]
> Is the above code the minimal code needed to cause the error? i.e. do you need to specify font-family and font.serif? I find needing these a bit strange...

If font.family is set to 'times' or 'Times', I can reproduce.  It doesn't seem to matter whether font.serif is specified.

This line is necessary (in my case) to match the publication font family.  

> Do other Greek symbols have a problem?

Yes, \rho and \sigma both cause me trouble.  I stopped testing there.

> Is it just the PDF backend w/ the problem or does svg and eps show this too?

PDF shows the problem.  SVG and EPS do not—they display as expected.

**Comment by jklymak** [comment]
Thanks a lot for checking @jkulesza

**Comment by jklymak** [comment]
Ah, maybe I don't get it because I can't find those font families:

```
/Users/jklymak/matplotlib/lib/matplotlib/font_manager.py:1220: UserWarning: findfont: Font family ['serif'] not found. Falling back to DejaVu Sans.
  (prop.get_family(), self.defaultFamily[fontext]))
```
and 
```
\Users/jklymak/matplotlib/lib/matplotlib/font_manager.py:1220: UserWarning: findfont: Font family ['times'] not found. Falling back to DejaVu Sans.
  (prop.get_family(), self.defaultFamily[fontext]))
```

**Comment by pharshalp** [comment]
> Can anyone produce this not on macOS? 

I tried this in a VM (Ubuntu 18.04) with --
Python 3.6.5 :: Anaconda custom (64-bit)
Matplotlib 2.2.2, pdfTeX 3.14159265-2.6-1.40.18 (TeX Live 2017/Debian)
kpathsea version 6.2.3


1.  with `mpl.rcParams['font.serif'] = ['times']` I get the following warning (missing font families).

anaconda3/lib/python3.6/site-packages/matplotlib/font_manager.py:1328: UserWarning: findfont: Font family ['serif'] not found. Falling back to DejaVu Sans

2.  with `mpl.rcParams['font.serif'] = ['times new roman']` I do not get any warning.

In both cases, the pdf looks fine (with no streaks).

**Comment by pdhall99** [comment]
You can make a more minimal example that produces the error if you only change the following rcParams:
```python
mpl.rcParams['text.usetex'] = True
mpl.rcParams['font.serif'] = ['times']
```

I find that changing
```python
mpl.rcParams['text.usetex'] = True
mpl.rcParams['font.family'] = ['times']
```
produces a PDF with all the text is Computer Modern, not Times. As does the following:
```python
mpl.rcParams['text.usetex'] = True
mpl.rcParams['font.family'] = ['times new roman']
```

A minimal example that avoids the warning that matplotlib can't find Times.ttf but still has the problem displaying alpha:
```python
import matplotlib as mpl
import matplotlib.pyplot as plt

mpl.rcParams.update(mpl.rcParamsDefault)
mpl.rcParams['text.usetex'] = True
mpl.rcParams['font.serif'] = ['times', 'times new roman']

fig, ax = plt.subplots()
ax.set_xlabel(r'$x$ $\alpha$')
fig.savefig('test.pdf')
```

**Comment by lboogaard** [comment]
Has there been any progress on this? I'm now running into this issue on osx after re-creating a figure that was fine before.

(Note that, as temporary workaround, one could save as a `png`.)

**Comment by pdhall99** [comment]
If you have the newtxmath package then another workaround that works with PDF output is to add
```python
mpl.rcParams['text.latex.preamble'] = [r'\usepackage{newtxmath}'] 
```

**Comment by astro313** [comment]
I have tried all the solutions above and they solve my problem if I plot and open them on my mac. But the problem persists when I open a file copied from a linux workstation. I've tried installing times.ttf (and make a copy to /anaconda2/lib/python2.7/site-packages/matplotlib/mpl-data/fonts/ttf) and also ~/Library/Fonts (and installing to the system), but the greek letters still appear as streaks in the copied file.

**Comment by jsalort** [comment]
>     * Can anyone produce this _not_ on macOS?

I reproduce it with a figure produced on Debian 4.9.144-3.1 and matplotlib 3.1.0/python 3.7.3 installed from anaconda. 

The figure shows correctly on the PDF Viewer opened by Jupyterlab in the web browser. However, when I save and open the file on the Mac with Preview.app, there is a streak. When I open it in Adobe Reader DC, I get an error message that it is impossible to extract the font "StandardSymL_Slant_167" and that some characters may not show properly. When the figure PDF in included in a LaTeX document, there are streaks.

**Comment by calvin-sykes** [comment]
I was also affected by this issue; somewhat embarrassingly, in a paper uploaded to arXiv.
The following Ghostscript command (from [here](https://www.karlrupp.net/2016/01/embed-all-fonts-in-pdfs-latex-pdflatex/)) resolved it for me:

`gs -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -dPDFSETTINGS=/prepress -dEmbedAllFonts=true -sOutputFile=my_plot_embedded.pdf -f my_plot.pdf`

I had to run this for each figure individually, running the command on the final LaTeX document did not work.

**Comment by markushartmann** [comment]
I reproduced also the error on Mac 10.12.6 mit python 2.7.17 und matplotlib 2.2.4.

Screenshot Adobe Reader

<img width="978" alt="Bildschirmfoto 2020-01-10 um 22 23 33" src="https://user-images.githubusercontent.com/10390818/72187719-bdf3d100-33f8-11ea-807b-a29289280dd8.png">

Screenshot Preview

<img width="850" alt="Bildschirmfoto 2020-01-10 um 22 24 15" src="https://user-images.githubusercontent.com/10390818/72187745-ccda8380-33f8-11ea-83ec-a9601de5c735.png">

**Comment by spmulders** [comment]
> I was also affected by this issue; somewhat embarrassingly, in a paper uploaded to arXiv.
> The following Ghostscript command (from [here](https://www.karlrupp.net/2016/01/embed-all-fonts-in-pdfs-latex-pdflatex/)) resolved it for me:
> 
> `gs -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -dPDFSETTINGS=/prepress -dEmbedAllFonts=true -sOutputFile=my_plot_embedded.pdf -f my_plot.pdf`
> 
> I had to run this for each figure individually, running the command on the final LaTeX document did not work.

This actually works, thanks! On Windows, use gswin64c instead of gs, contained in the bin folder of Ghostscript.

**Comment by jkseppan** [comment]
I think this could be the same bug as #16087, which should be fixed on current master (by PR #18080).

**Comment by QuLogic** [comment]
I confirmed on the Mac mini that this is fixed by #18080.
